Add iOS notifications

elijahverdoorn opened the pull request:
This PR adds notification support on iOS. There's a lot of files in the PR that are added by the library that I added from OneSignal to make this happen, but all the changes that we care about are in AppDelegate.m, AppDelegate.h, and index.ios.js on the javascript side. All the rest of it is just config changes to get the new library in there.

hawkrives commented on ios/AllAboutOlaf.xcodeproj/project.pbxproj:
```diff
@@ -749,15 +897,13 @@
-				BUNDLE_LOADER = "$(TEST_HOST)";
 				COPY_PHASE_STRIP = NO;
-				INFOPLIST_FILE = AllAboutOlafTests/Info.plist;
+				INFOPLIST_FILE = /Users/verdoo1/Desktop/aao/ios/AllAboutOlafTests/Info.plist;
```
We'll need to manually change these back to relative paths.

elijahverdoorn replied:
Boo... Stupid Xcode.

elijahverdoorn replied:
Done.

hawkrives commented on ios/AllAboutOlaf.xcodeproj/project.pbxproj:
```diff
@@ -783,13 +932,14 @@
 					"-lc++",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = "org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)";
+				PRODUCT_BUNDLE_IDENTIFIER = "NFMTHAZVS9.com.drewvolz.stolaf-";
```
Did this need to change? Shouldn't it still include the PRODUCT_NAME variable?

elijahverdoorn replied:
This does need to change to match what Apple has on file for the app since we are publishing as an update rather than a new app.

hawkrives commented on ios/AllAboutOlaf/Info.plist:
```diff
@@ -1,117 +1,121 @@
+	<key>UIViewControllerBasedStatusBarAppearance</key>
+	<false/>
+</dict>
 </plist>
```
i guess we can leave these whitespace changes… grumble.

hawkrives commented on views/streaming/radio.js:
```diff
@@ -13,7 +13,7 @@ import {
 // let kstoDownload = 'itms://itunes.apple.com/us/app/ksto/id953916647'
 
-const url = 'http://pages.stolaf.edu/ksto/listen/'
+const url = 'https://pages.stolaf.edu/ksto/listen/'
```
Please remove this change. We've already merged it into master once :)

elijahverdoorn committed:
Remove ride-along change

Didn't intend for that to be in here, was just faster to make another commit than to revert.

elijahverdoorn committed:
Use relative paths

Xcode used absolute paths. I want to use relative paths. I am smarter than Xcode.

hawkrives commented on ios/AllAboutOlaf.xcodeproj/project.pbxproj:
```diff
@@ -462,6 +499,17 @@
 						TestTargetID = 13B07F861A680F5B00A75B9A;
 					};
+					13B07F861A680F5B00A75B9A = {
+						DevelopmentTeam = 363G7T9TGX;
```
This might prevent me from building locally. I'll check tonight.

hawkrives committed:
Make Travis work with Cocoapods (#164)

* Change the test command to use the workspace file

* switch to xcodebuild

* Revert "switch to xcodebuild"

fc5fa3e8fc3e852598227be56ebdabc07866b10b

* switch to just building iOS

hawkrives committed:
Add iOS notifications

Using Cocoapods to add the code driving the OneSignal service for the notificaitons. The backend is registering the simulator that I'm using for iOS testing, but since simulators cannot get notificaitons I haven't seen the end-to-end process work yet. I'll be able to do that within the day and will make an update if it's not working as expected.

hawkrives committed:
missed a thing while rebasing
